toolbox/relnotes/main.go

```diff
@@ -0,0 +1,606 @@
+		if err != nil {
+			log.Printf("failed to get CI status: %s", err)
+			return
+		}
+	}
+
+	if *htmlFileName != "" {
+		err = createHTMLNote(*htmlFileName, *mdFileName)
+		if err != nil {
+			log.Printf("failed to generate HTML release note: %s", err)
+		}
+	}
+
+	return
+}
+
+// getPendingPRs gets pending PRs on given branch in the repo.
+func getPendingPRs(c *github.Client, f *os.File, owner, repo, branch string) error {
+	log.Printf("Adding pending PR status...")
+	f.WriteString("-------\n")
+	f.WriteString(fmt.Sprintf("## PENDING PRs on the %s branch\n", branch))
+
+	if *htmlizeMD {
+		f.WriteString("PR | Milestone | User | Date | Commit Message\n")
+		f.WriteString("-- | --------- | ---- | ---- | --------------\n")
+	}
+
+	var query []string
+	query = u.AddQuery(query, "repo", owner, "/", repo)
+	query = u.AddQuery(query, "is", "open")
+	query = u.AddQuery(query, "type", "pr")
+	query = u.AddQuery(query, "base", branch)
+	pendingPRs, err := u.SearchIssues(c, strings.Join(query, " "))
+	if err != nil {
+		return fmt.Errorf("failed to search pending PRs: %s", err)
+	}
+
+	for _, pr := range pendingPRs {
+		var str string
+		// escape '*' in commit messages so they don't mess up formatting
+		msg := strings.Replace(*pr.Title, "*", "", -1)
+		milestone := "null"
+		if pr.Milestone != nil {
+			milestone = *pr.Milestone.Title
+		}
+		if *htmlizeMD {
+			str = fmt.Sprintf("#%-8d | %-4s | @%-10s| %s   | %s\n", *pr.Number, milestone, *pr.User.Login, pr.UpdatedAt.Format("Mon Jan  2 15:04:05 MST 2006"), msg)
+		} else {
+			str = fmt.Sprintf("#%-8d  %-4s  @%-10s %s    %s\n", *pr.Number, milestone, *pr.User.Login, pr.UpdatedAt.Format("Mon Jan  2 15:04:05 MST 2006"), msg)
+		}
+		f.WriteString(str)
+	}
+	f.WriteString("\n\n")
+	return nil
+}
+
+// createHTMLNote generates HTML release note based on the input markdown release note.
+func createHTMLNote(htmlFileName, mdFileName string) error {
+	cssFileName := "/tmp/release_note_cssfile"
+	cssFile, err := os.Create(cssFileName)
+	if err != nil {
+		return fmt.Errorf("failed to create css file %s: %s", cssFileName, err)
+	}
+
+	cssFile.WriteString("<style type=text/css> ")
+	cssFile.WriteString("table,th,tr,td {border: 1px solid gray; ")
+	cssFile.WriteString("border-collapse: collapse;padding: 5px;} ")
+	cssFile.WriteString("</style>")
+	cssFile.Close()
+
+	htmlStr, err := u.Shell("pandoc", "-H", cssFileName, "--from", "markdown_github", "--to", "html", mdFileName)
+	if err != nil {
+		return fmt.Errorf("failed to generate html content: %s", err)
+	}
+
+	htmlFile, err := os.Create(htmlFileName)
+	if err != nil {
+		return fmt.Errorf("failed to create html file: %s", err)
+	}
+	defer htmlFile.Close()
+
+	htmlFile.WriteString(htmlStr)
+	return nil
+}
+
+// getCIJobStatus runs the script find_green_build and append CI job status to outputFile.
+func getCIJobStatus(outputFile, branch string, htmlize bool) error {
+	log.Printf("Adding CI job status (this may take a while)...")
```

Comment:
I thought it's "getting" instead of "adding"

Reply:
Done.